src/utils/time.ts

```diff
@@ -91,3 +91,13 @@ export function toMilliseconds(components: TimeComponents) {
 export function toSeconds(components: TimeComponents) {
     return toMilliseconds(components) / 1000;
 }
+
+/**
+ * Sleeps for a several amount of time
+ * @param milliseconds
+ */
+export function sleep(milliseconds: number) {
```

Comment:
You can just call `sleep(toMilliseconds(components))`

Reply:
Yes, of course, but now that it's a public API function and specifically part of an API that defines strongly typed durations, it certainly seems odd to me that it would not actually use them. E.g. I can just as well do `sleep(toSeconds(components))` - it would just be wrong.

Reply:
(And even more to the point: _any conversion_ not `toMilliseconds` would be possible-but-wrong. There is only exactly this one which is correct, so putting it directly into `sleep` seems to be a good idea.)